stingray/powerspectrum.py

```diff
@@ -399,3 +399,77 @@ def _make_segment_spectrum(self, lc, segment_size):
             nphots_all.append(np.sum(lc_seg.counts))
 
         return power_all, nphots_all
+
+
+class DynamicalPowerspectrum(AveragedPowerspectrum):
+    def __init__(self, lc, segment_size, norm="frac", gti=None):
+        """
+        Parameters
+        ----------
+        lc : lightcurve.Lightcurve object
+            The time series of which the Dynamical powerspectrum is
+            to be calculated.
+
+        segment_size : float, default 1
+             Length of the segment of light curve, default value is 1 second.
+
+        norm: {"leahy" | "frac" | "abs" | "none" }, optional, default "frac"
+            The normaliation of the periodogram to be used. Options are
+            "leahy", "frac", "abs" and "none", default is "frac".
+
+        Attributes
+        ----------
+        dyn_ps : np.ndarray
+            The matrix with fourier frequencies in the first column and time
+            in the second column.
+        """
+        if segment_size < 2*lc.dt:
+            raise ValueError("Length of the segment is too short to form a "
+                             "light curve!")
+        elif segment_size > lc.tseg:
+            raise ValueError("Length of the segment is too long to create "
+                             "any segments of the light curve!")
+        self.segment_size = segment_size
+        self.norm = norm
+        self.gti = gti
+        self.power_all, _ = AveragedPowerspectrum._make_segment_spectrum(
+            self, lc, segment_size)
+        self._make_matrix()
+
+    def _make_matrix(self):
+        """Create the matrix with freq and time columns."""
+        row1 = self.power_all[0].freq
+        row2 = np.arange(0, self.segment_size*len(row1), self.segment_size)
```

Comment:
I believe row2 definition has to be:

`row2 = np.arange(0, lc.tseg, self.segment_size)`

This will give the time-stamps of the segments starting from 0 up to the total lenght of the light curve (in units of time) separated by `segment_size`

A decision has to be made here to use interval time (from 0 to lc.tseg) or the actual time unit of the lighcurve
(from `lc.time[0] -  0.5*lc.dt + 0.5*segment_size` to `lc.time[-1] + 0.5*lc.dt` , in this case the limits of `np.arange` need to be changed accordingly). Note that in any case it will be convenient to pass `lc` as an argument to `_make_matrix`

My vote goes for the second case, using the time units of the lightcurve.